doc/qos/new-headroom-calculation.md

```diff
@@ -0,0 +1,168 @@
+# Requirement
+## Current Solution
+
+- When system start, it reads the pg_profile_lookup.ini and generate an internal lookup table indexed by speed and cable length, and containing size, xon, xoff and threshold.
+- When a port's cable length updated, it records the cable length of the port. But it doesn't update relavent tables accordingly.
+- When a port's speed updated,
+    1. It looks the (speed, cable length) pair up in the BUFFER_PROFILE table or generate a new profile according to the internal lookup table.
+    2. And then update the port's BUFFER_PG table for the lossless priority group.
+
+## The Requirement
+We will have the following groups of parameters
+- List of SONiC configuration (such as speed and cable length)
+- List of pg configuration (such as cell size, IPG, MTU, Peer response time, small packet size percentage)
+
+Based on the parameters and a well-known formula the code in buffer manager will do the calculation and not take it from a pre-defined values as we have today.
+On top of that, we need to support the ability to override headroom and not to calculate it in the code.
+# Design
+## The overall flow
+
+ - When system starts, load the SWITCH_PARAMETER table or the pg_profile_lookup.ini.

```

Comment:
I thought we are removing "pg_profile_lookup.ini"?

Reply:
But other vendors probably aren't able to provide SWITCH_PARAMETERs in one day.
Let's keep it in order to tolerance vendors using the old approach to calculating the headroom.